scripts/veloxbench/veloxbench/cpp_micro_benchmarks.py

```diff
@@ -0,0 +1,121 @@
+import copy
+import json
+import os
+import tempfile
+
+import conbench.runner
+
+from veloxbench import _benchmark
+
+RUN_OPTIONS = {
+    "iterations": {
+        "default": None,
+        "type": int,
+        "help": "Number of iterations of each benchmark.",
+    },
+}
+
+
+# TODO: what other COMMON_OPTIONS?
+# https://github.com/ursacomputing/benchmarks/blob/033eee0951adbf41931a2de95caccbac887da6ff/benchmarks/cpp_micro_benchmarks.py#L18-L59
+# TODO: the suite filter won't quite work yet either
+COMMON_OPTIONS = {
+    "suite-filter": {
+        "default": None,
+        "type": str,
+        "help": "Regex filtering benchmark suites.",
+    },
+}
+
+
+def get_run_command(filename, options):
+    # TODO: we'll probably want to unify these runner scripts
+    # Must run `make benchmarks-basic-build` before this
+    # TODO: deal with the paths better (especially to benchmarks/basic)
+    command = [
+        "../../../scripts/benchmark-runner.py",
+        "run",
+        "--path",
+        "../../../_build/release/velox/benchmarks/basic/",
+        "--dump-path",
+        filename,
+    ]
+
+    # TODO: extend cpp micro benchmarks to allow for iterations
+    iterations = options.get("iterations", None)
+    if iterations:
+        raise NotImplementedError()
+
+    return command
+
+
+def _parse_benchmark_name(full_name):
+    # TODO: Do we need something more complicated?
+    # https://github.com/ursacomputing/benchmarks/blob/033eee0951adbf41931a2de95caccbac887da6ff/benchmarks/cpp_micro_benchmarks.py#L86-L103
+    return {"name": full_name}
+
+
+@conbench.runner.register_benchmark
+class RecordCppMicroBenchmarks(_benchmark.Benchmark):
+    """Run the Velox C++ micro benchmarks."""
+
+    external = True
+    name = "cpp-micro"
+    # see TODO about COMMON_OPTIONS above, set an empty dict for now
+    options = copy.deepcopy(COMMON_OPTIONS)
+    options.update(**RUN_OPTIONS)
+    description = "Run the Velox C++ micro benchmarks."
+    iterations = 1
+    flags = {"language": "C++"}
+
+    def run(self, **kwargs):
+        with tempfile.TemporaryDirectory() as result_dir:
+            run_command = get_run_command(result_dir, kwargs)
+            self.execute_command(run_command)
+            # iterate through files to make the suites
+            with os.scandir(result_dir) as result_files:
+                for result_file in result_files:
+                    suite = result_file.name.removesuffix(".json")
+                    with open(result_file.path, "r") as f:
+                        results = json.load(f)
+                    self.conbench.mark_new_batch()
+                    for result in results:
+                        yield self._record_result(suite, result, kwargs)
+
+    def _record_result(self, suite, result, options):
+        info, context = {}, {"benchmark_language": "C++"}
+        # TODO: we should really name these in the json and not rely on position!
+        tags = _parse_benchmark_name(result[1])
```

Comment:
It would be nice if the output json for the benchmarks were dictionaries | named lists instead of just lists. We also might want to expand the details of what is being collected, for Arrow, we collect iterations per second (sometimes) as well as elapsed time. Along with details about if a lower or higher number is better, etc.